### PR TITLE
Make `FetchPhase` logic more readable. 

### PR DESCRIPTION
**Pull request by jtibshirani**
* Factor out `FieldsVisitor#postProcess` call.
* Swap logical order for normal and nested documents.
* Extract the method `createStoredFieldsVisitor`.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java [hunk 1]
```diff
@@ -209,6 +163,54 @@ public int compareTo(DocIdToIndex o) {
         }
     }
 
+    private FieldsVisitor createStoredFieldsVisitor(SearchContext context, Map<String, Set<String>> storedToRequestedFields) {
```
Comment (mayya-sharipova) [inline server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java:166]: do we want to make this function static?

File: server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java [hunk 2]
```diff
@@ -438,13 +437,17 @@ private void prepareNestedHitContext(FetchSubPhase.HitContext hitContext,
         return nestedIdentity;
     }
 
-    private void loadStoredFields(SearchShardTarget shardTarget, LeafReaderContext readerContext, FieldsVisitor fieldVisitor, int docId) {
+    private void loadStoredFields(SearchShardTarget shardTarget,
```
Comment (mayya-sharipova) [inline server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java:440]: do we want to make this function `static`?

**Review by mayya-sharipova — APPROVED** [review]
Thanks @jtibshirani, indeed the code became more readable.

File: server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java (hunk 1 above)
Comment (jtibshirani) [reply server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java:166]: 👍

File: server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java (hunk 2 above)
Comment (jtibshirani) [reply server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java:440]: 👍

File: server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java (hunk 1 above)
Comment (jtibshirani) [reply server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java:166]: On second thought, would you be okay keeping these as instance methods for consistency? It looks like all the other helper methods (`findRootDocumentIfNested`, `prepareHitContext`, etc.) are not static.

Comment (mayya-sharipova) [reply server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java:166]: I don't mind keeping them instance methods for consistency.

**Pull request merged**